### PR TITLE
Optimized data structures

### PR DESCRIPTION
**Pull request by young24**
Hi Wanzhuo, 
I have optimized the data structures of BLITZ, mostly the WriteOutData class.
Once you come back, could you please do the final test to make all things run as we design? Thanks in advance. 
@WanzhuoShi

### CODE REVIEW

**Comment by WanzhuoShi** [comment]
Sure, I will test this branch immediately.

File: MyLibs/talk2screen.cpp [hunk 1]
```diff
@@ -251,52 +308,17 @@ bool ScreenData::init_glad()
...
-		}
-		stbi_image_free(data[i]);
-	}
-	return true;
-}
-
 /* Render designed pattern on the screen */
 void ScreenData::renderTexture()
-{
+{// TODO: use functions to replace loops
```
Comment (young24) [inline MyLibs/talk2screen.cpp]: You can work on this function when you have time. @WanzhuoShi

File: MyLibs/writeOutFish.h [hunk 2]
```diff
@@ -25,6 +25,13 @@
 * Created on: Jan. 1, 2018
 */
 
+// TODO: redefine this class, think it clear the relationship between
+// this WriteOut class and the ExperimentData class
+// TODO: unify the input from one place (file, script or command window)
```
Comment (young24) [inline MyLibs/writeOutFish.h]: This was done by you. Good job. Now all the inputs are required through the command window. 
@WanzhuoShi 👍

File: MyLibs/talk2screen.cpp (hunk 1 above)
Comment (WanzhuoShi) [reply MyLibs/talk2screen.cpp]: Okay, I will try it tomorrow.

File: MyLibs/writeOutFish.h [hunk 3]
```diff
@@ -25,6 +25,13 @@
...
+// TODO: create a yaml config file and corresponding functions
+// TODO: rewrite the WriteOutData class as the FileIO class
```
Comment (young24) [inline MyLibs/writeOutFish.h]: How do you think about this point? Is it necessary to recreate this writeOut class as the FileIO class? 
@WanzhuoShi

**Pull request merged**